runtime/databricks/automl_runtime/forecast/pmdarima/training.py

```diff
@@ -106,4 +110,18 @@ def _fit_predict(df: pd.DataFrame, cutoffs: List[pd.Timestamp], seasonal_period:
     @staticmethod
     def _fill_missing_time_steps(df: pd.DataFrame, frequency: str):
         # Forward fill missing time steps
-        return df.set_index("ds").resample(rule=OFFSET_ALIAS_MAP[frequency]).pad().reset_index()
+        df_filled = df.set_index("ds").resample(rule=OFFSET_ALIAS_MAP[frequency]).pad().reset_index()
+        start_ds, modified_start_ds = df["ds"].min(), df_filled["ds"].min()
+        if start_ds != modified_start_ds:
+            offset = modified_start_ds - start_ds
+            df_filled["ds"] = df_filled["ds"] - offset
+        return df_filled
+
+    @staticmethod
+    def _validate_ds_freq(df: pd.DataFrame, frequency: str):
```

Comment:
The check in `ArimaModel._predict_impl` and here should ideally use a common function. Will address it in the clean up ticket.
Same for the `start_ds` and `modified_start_ds` check in `ArimaEstimator._fill_missing_time_steps` and `AbstractArimaModel._get_ds`.